module-audio/Audio/decoder/Decoder.cpp

```diff
@@ -50,17 +48,19 @@ namespace audio
         return std::make_unique<tags::fetcher::Tags>(tags::fetcher::fetchTags(filePath));
     }
 
-    std::unique_ptr<Decoder> Decoder::Create(const char *file)
+    std::unique_ptr<Decoder> Decoder::Create(const std::string &filePath)
     {
+        const auto filePathLowercase = utils::stringToLowercase(filePath);
         std::unique_ptr<Decoder> dec;
-        if ((strstr(file, ".wav") != NULL) || (strstr(file, ".WAV") != NULL)) {
-            dec = std::make_unique<decoderWAV>(file);
+
+        if (filePathLowercase.find(".wav") != std::string::npos) {
```

Comment:
That's not a robust way to check the file extension. The file name could be e.g. _ocean.**wav**es.sound.mp3_ and the file will be detected as a WAVE, not an MP3.

Reply:
You're right, thanks for spotting this case! Fixed ;)